src/query/pipeline/sources/src/input_formats/source_deserializer.rs

```diff
@@ -111,7 +111,9 @@ impl<I: InputFormatPipe> Processor for DeserializeSource<I> {
         }
         let blocks = self.block_builder.deserialize(self.input_buffer.take())?;
         for b in blocks.into_iter() {
-            self.output_buffer.push_back(b)
+            if !b.is_empty() {
```

Comment:
row batches from diff files may be delivered to the same  DeserializeProcessor or DeserializeSource, and merge into one block, which should be forbiddened when use skip_file_num,  is this handled in this  pr?

Reply:
Yup, handled here: https://github.com/datafuselabs/databend/pull/9793/files#diff-23dfe539c8b7d8c6d471f64802aa6377c323cb2465c59448e5639d1d91a7111bR544